### ISSUE TITLE
Feature idea - watch folder sizes

### ISSUE THREAD

**Issue by ofirgalcon** [munkireport/munkireport-php#1043]
Allow to define folder(s) to be scanned daily for size and report back. One folder I would check is /var/log where some poorly behaved apps leave gigantic logs.

**Comment by flammable** [comment]
You can track storage usage in Listings > Storage.  Although it doesn't break down where the space is going, you can use tools like [OmniDiskSweeper](https://www.omnigroup.com/more) or [GrandPerspective](http://grandperspectiv.sourceforge.net) to narrow that down for the user (I have both available in MSC for my org).

**Comment by ofirgalcon** [comment]
These are useful but requires users to act, my users don't act.

**Comment by flammable** [comment]
Mine don't either.  I reach out to them when MunkiReport shows that installations are failing due to lack of disk space.

**Comment by ofirgalcon** [comment]
Me too, they don't reply.

**Comment by flammable** [comment]
So what would you do with this information, then?

**Comment by ofirgalcon** [comment]
Run a script to deal with the problem if possible.

**Comment by flammable** [comment]
By deleting an arbitrarily large amount of data? You'd script that?

**Comment by ofirgalcon** [comment]
Totally. I found that LogMeIn and MalwareBytes had bugs that created huge logs. My script zipped the log and restarted it. a 5GB file reduced to about 50MB.

**Comment by tuxudo** [comment]
I've thought about doing a module like this, but the problem is I haven't found a quick way to get a folder size without causing performance problems or timing out in preflight. I think APFS has a quick method for this, but HFS+ does not

**Comment by ofirgalcon** [comment]
It would have to run as a separate process I guess and with low io priority, scan less often, etc. then save results to a file MR module would read.

**Comment by danner26** [comment]
why not have the module do a du, and if it is above the file size specified, then set a cronjob to remove the file or do whatever action 

just a shot, may or may not work

**Comment by danner26** [comment]
or like you said create a separate program and integrate reporting to MR

**Comment by ofirgalcon** [comment]
In answer to your question - this module would help discovering such issues. It took me a  while to realise something was eating away at storage space.

**Comment by ofirgalcon** [comment]
APFS support would be a start.

**Comment by danner26** [comment]
no to the du/cronjob option? lol

**Comment by ofirgalcon** [comment]
My idea was a module that would help discover the issue in the first place, and I already have a munki nopkg script running doing pretty much what you suggested.

**Comment by danner26** [comment]
oh i see, you're not worried about files you know are large, you're worried about ones you do not know about.. well how would you script that and not delete an important file?? or do you just want to report the largest files?

**Comment by ofirgalcon** [comment]
Exactly, I need to discover balloons. So I have a list of likely folders to watch. Logs Caches etc.

Once I see a folder is large I investigate.
